ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/DiskPolicy.hs

```diff
@@ -96,15 +88,36 @@ defaultDiskPolicy (SecurityParam k) requestedInterval = DiskPolicy {..}
          Maybe DiffTime
       -> Word64
       -> Bool
-    onDiskShouldTakeSnapshot Nothing blocksSinceLast = blocksSinceLast >= k
     onDiskShouldTakeSnapshot (Just timeSinceLast) blocksSinceLast =
-      let snapshotIntervalSeconds      = fromIntegral $ interval requestedInterval
-          snapshotInterval             = secondsToDiffTime snapshotIntervalSeconds
-          itsBeen50kBlocks             = blocksSinceLast >= 50_000
-          itsBeen6MinSinceLastSnapshot = timeSinceLast > 6 * secondsToDiffTime 60
-      in
-           timeSinceLast >= snapshotInterval
-        || itsBeen50kBlocks && itsBeen6MinSinceLastSnapshot
+         timeSinceLast >= snapshotInterval
+      || substantialAmountOfBlocksWereProcessed blocksSinceLast timeSinceLast
+
+   -- | If users never leave their wallet running for long. This would mean
+   -- that under some circumstances we would never take a snapshot
+   -- So, on startup (when the 'time since the last snapshot' is `Nothing`),
+   -- we take a snapshot as soon as there are @k@ blocks replayed.
+   -- This means that even if users frequently shut down their wallet, we still
+   -- take a snapshot roughly every @k@ blocks. It does mean the possibility of
+   -- an extra unnecessary snapshot during syncing (if the node is restarted), but
+   -- that is not a big deal.
+    onDiskShouldTakeSnapshot Nothing blocksSinceLast = blocksSinceLast >= k
+
+    -- | We want to create a snapshot after a substantial amount of blocks were
+    -- processed (hard-coded to 50k blocks). Given the fact that during bootstrap
+    -- a fresh node will see a lot of blocks over a short period of time, we want
+    -- to limit this condition to happen not more often then a fixed amount of
+    -- time (here hard-coded to 6 minutes)
+    substantialAmountOfBlocksWereProcessed blocksSinceLast timeSinceLast =
+      let minBlocksBeforeSnapshot      = 50_000
```

Comment:
Style: drop the extra spaces before `=`, unless you think they make it easier to read (I'm guessing they're just holdovers from the previous layout.)

Reply:
It's for readability 

```
    substantialAmountOfBlocksWereProcessed blocksSinceLast timeSinceLast =
      let minBlocksBeforeSnapshot      = 50_000
          minTimeBeforeSnapshot        = 6 * secondsToDiffTime 60
      in    blocksSinceLast >= minBlocksBeforeSnapshot
         && timeSinceLast   >= minTimeBeforeSnapshot
```